openedx-unsupported/configuration
# Add variable NOTIFIER_POLLING_INTERVAL

Configuration Pull Request
---

This adds the possibility of changing the interval at which the celery worker in notifier refreshes the connection to the database.
- Before this change, it was using [the default defined by kombu](https://github.com/celery/kombu/blob/249e56525c17ece5791ebb283fa2679d6155986f/kombu/transport/django/__init__.py#L28) which is 5 seconds, which we found too fast.
- After this change, there's a new ansible variable `NOTIFIER_POLLING_INTERVAL` (default 5) to specify this interval; this ends up in a Django setting `KOMBU_POLLING_INTERVAL` and then in kombu's `POLLING_INTERVAL`.
- To do this change, notifier now uses the [settings_local.py](https://github.com/edx/configuration/blob/master/playbooks/roles/notifier/templates/settings_local.py.j2) settings file (which extends `settings.py`). Before this change, this settings file was used only when there was a notifier theme; now it's always used.  

**JIRA tickets**: None
**Discussions**: None
**Dependencies**: None
**Screenshots**: None
**Sandbox URL**: None
**Merge deadline**: None

**Testing instructions**:

1. Deploy a server with this playbook and a variable `NOTIFIER_POLLING_INTERVAL = "60"`
2. It should deploy
3. Connect to the database server (logins at `/edx/app/notifier/notifier_env`), and run a command like `show full processlist` (MySQL). In the `time` column it shows the connection age. Before this change, you never saw values greater 5. After this change, they go until the number of seconds you chose (60).
4. Everything else should work. In particular, what notifier does (daily digests of new content to subscribed forums users). If you subscribe to forum digests, then you can either wait for some days to see if you receive digests, or (this is a bit tricky) from inside the notifier virtualenv and after you define the variables found in the supervisor conf, you can run a command like `cd /edx/app/notifier/src$ /edx/app/notifier/virtualenvs/notifier/bin/python manage.py forums_digest --minutes 1 --show-users  -v 3` and see that the notifier code is running and finding users  
5. Verify the modified files: `/edx/app/supervisor/conf.d/notifier-celery-workers.conf`, `/edx/app/notifier/src/notifier/settings_local.py`

**Reviewers**

**Author concerns**: None
**Settings**: None


Make sure that the following steps are done before merging:
